File: NutritionControl.DTO/Models/ReceiptDto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NutritionControl.DTO.Models
{
    public class ReceiptDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string PhotoUrl { get; set; }
        public string Description { get; set; }
        public decimal Calories { get; set; }
        public string CategoryName { get; set; }
        public string Method { get; set; }
        public List<ProductReceiptDto> Products { get; set; }
    }
    public class ProductReceiptDto
    {
        public double Count { get; set; }
        public string Measurment { get; set; }
        public ProductDto Product { get; set; }
    }
}

File: NutritionControl.Client/client-app/src/app/models/receiptDto.ts
import { ProductDto } from './productDto'

export class ProductReceiptDto {
  count: number
  measurment: string
  product: ProductDto
}

export class ReceiptDto {
  id: number
  name: string
  photoUrl: string
  description: string
  calories: number
  protein:number
  fats:number
  carbohydrates:number
  categoryName: string
  method: string
  products: Array<ProductReceiptDto>
}